platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/roster/RosterUtilsTest.java

```diff
@@ -17,29 +17,154 @@
+import com.swirlds.platform.system.SoftwareVersion;
+import com.swirlds.platform.system.address.Address;
+import com.swirlds.platform.system.address.AddressBook;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
-public class RosterUtilsTest {
+class RosterUtilsTest {
+
+    private final AddressBook addressBook = mock(AddressBook.class);
+    private final Address address = mock(Address.class);
+    private final NodeId nodeId = new NodeId(1);
+    private final X509Certificate certificate = mock(X509Certificate.class);
+
     @Test
     void tesetHash() {
-        final Hash hash = RosterUtils.hash(Roster.DEFAULT);
+        final Hash hash = RosterUtils.hashOf(Roster.DEFAULT);
         assertEquals(
                 "38b060a751ac96384cd9327eb1b1e36a21fdb71114be07434c0cc7bf63f6e1da274edebfe76f65fbd51ad2f14898b95b",
                 hash.toString());
 
-        final Hash anotherHash = RosterUtils.hash(
+        final Hash anotherHash = RosterUtils.hashOf(
                 Roster.DEFAULT.copyBuilder().rosterEntries(RosterEntry.DEFAULT).build());
         assertEquals(
                 "5d693ce2c5d445194faee6054b4d8fe4a4adc1225cf0afc2ecd7866ea895a0093ea3037951b75ab7340b75699aa1db1d",
                 anotherHash.toString());
 
-        final Hash validRosterHash = RosterUtils.hash(RosterValidatorTests.buildValidRoster());
+        final Hash validRosterHash = RosterUtils.hashOf(RosterValidatorTests.buildValidRoster());
         assertEquals(
                 "1b8414aa690d96ce79e972abfc58c7ca04052996f89c5e6789b25b9051ee85fccb7c8ed3fc6ebacef177adfdcbbb5709",
                 validRosterHash.toString());
     }
+
+    @Test
+    @DisplayName("Test construct initial roster with software upgrade")
+    void testConstructInitialRosterWithSoftwareUpgrade() {
+        final SoftwareVersion version = mock(SoftwareVersion.class);
+        final ReservedSignedState initialState = mock(ReservedSignedState.class);
+        final SignedState state = mock(SignedState.class);
+        final MerkleRoot stateMerkleRoot = mock(MerkleRoot.class);
+        final WritableRosterStore rosterStore = mock(WritableRosterStore.class);
+        final Roster candidateRoster = mock(Roster.class);
+        final PlatformStateAccessor platformState = mock(PlatformStateAccessor.class);
+
+        when(initialState.get()).thenReturn(state);
+        when(state.getState()).thenReturn(stateMerkleRoot);
+        when(state.getState().getWritableRosterStore()).thenReturn(rosterStore);
+        when(stateMerkleRoot.getReadablePlatformState()).thenReturn(platformState);
+        when(state.getState().getWritablePlatformState()).thenReturn(platformState);
+        when(rosterStore.getCandidateRoster()).thenReturn(candidateRoster);
+
+        assertNotNull(RosterUtils.constructInitialRoster(version, initialState, addressBook));
```

Comment:
This assertion is not sufficient and does not verify the intended behavior. The roster must match the candidate roster.